app/models/spree/user.rb

```diff
@@ -113,6 +113,12 @@ def associate_customers
       self.customers = Customer.where(email: email)
     end
 
+    def associate_orders
+      Spree::Order.placed_by_customer_email(email).each do |order|
+        order.associate_user!(self)
+      end
```

Comment:
I just noticed that the `where` clause is the same as in `associate_customers`. So this could be a lot simpler and faster:
```suggestion
      Spree::Order.where(customer: customers).find_each do |order|
        order.associate_user!(self)
      end
```

We don't need the scope and we don't need to join. Sorry for being a bit picky here but the less code the less maintenance for our future selves.